overlord/snapstate/snapstate.go

```diff
@@ -166,10 +169,19 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 
 	var prepare, prev *state.Task
 	fromStore := false
+
 	// if we have a local revision here we go back to that
 	if snapsup.SnapPath != "" || revisionIsLocal {
 		prepare = st.NewTask("prepare-snap", fmt.Sprintf(i18n.G("Prepare snap %q%s"), snapsup.SnapPath, revisionStr))
 	} else {
+		// XXX: DownloadInfo should never be nil, except for missing mocking in tests?
+		if snapsup.DownloadInfo != nil {
+			// require 10% extra + 1Mb
+			requiredSpace := uint64(snapsup.DownloadInfo.Size) + uint64(0.1 * float64(snapsup.DownloadInfo.Size)) + 1024*1024
```

Comment:
maybe it'd be better to extract this into a helper, eg
```
requiredSpace := requiredSpaceWithMargin(snapsup.DownloadInfo.Size)
```
otherwise it's buried deep down in the `if`s